VectorRumble.Shared/Gameplay/ShipManager.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using Microsoft.Xna.Framework.Content;
+using Microsoft.Xna.Framework.Content.Pipeline.Serialization.Intermediate;
+
+namespace VectorRumble
+{
+    internal class ShipManager 
+    {
+        const string shipDirectory = "Ships";
+        private readonly World world;
+
+        public ShipManager(World world)
+        {
+            this.world = world;
+        }
+
+        public List<Ship> Ships { get; set; } = new List<Ship>();
+        public Ship[] Players => Ships.Where(s => s.Playing).ToArray();
```

Comment:
does this need to be an Array? Can it be something else where we don't need to call `ToArray` ? I ask cos we are calling `ToArray` on this same result elsewhere in the code.

Reply:
`Ships` is now a dynamically sizing array, who's size is set when loading the ship files on startup. We aren't supporting ships being dynamically loaded, so that should be ok. I am using more Linq for OrderBy etc, which will need optimising shortly.

Reply:
I'm more concerned about performance. ToArray is quite expensive, as it all LINQ.

Reply:
@dellis1972 I was thinking about it, the only other alternative would be to have some methods that return `Players ` etc. So the question is would a method call that clears and repopulates a `List<Ship>` be quicker than that call to `ToArray()`? Have the .NET team not optimised the compiler for that scenario?

Reply:
The only thing that really needed optimising was the `SelectedShips` because it is used within the gameplay loop. Other ship properties are used within `ShipSelections` screen, so I don't think `ToArray` there a performance issue. We now have a `SelectedShip` property that is only updated outside the gameplay screen.